fml/concurrent_message_loop.cc

```diff
@@ -43,33 +48,18 @@ size_t ConcurrentMessageLoop::GetWorkerCount() const {
   return worker_count_;
 }
 
-std::shared_ptr<ConcurrentTaskRunner> ConcurrentMessageLoop::GetTaskRunner() {
-  return std::make_shared<ConcurrentTaskRunner>(weak_from_this());
-}
-
 void ConcurrentMessageLoop::PostTask(const fml::closure& task) {
   if (!task) {
     return;
   }
 
-  std::unique_lock lock(tasks_mutex_);
-
-  // Don't just drop tasks on the floor in case of shutdown.
-  if (shutdown_) {
-    FML_DLOG(WARNING)
-        << "Tried to post a task to shutdown concurrent message "
-           "loop. The task will be executed on the callers thread.";
-    lock.unlock();
-    task();
-    return;
-  }
```

Comment:
Tasks could always be dropped when shutting down a ConcurrentMessageLoop, there was no guarantee that they would all be executed despite a few places that attempted to make that a thing.  Specifically if more than N tasks were queued up and the loop was deleted, task[N+1, ..] would be dropped.

Dropping late tasks matches he behavior elsewhere in the engine (for example platform channel messages).

Reply:
Mainly notes for myself:

- This is ok because we don't give these workers to Dart. We only use them internally for SKP warmup and for image decoding.